src/EditorFeatures/Core/IntelliSense/AsyncCompletion/CompletionSource.cs

```diff
@@ -124,8 +121,8 @@ internal sealed class CompletionSource : IAsyncExpandingCompletionSource
                 // There could be mixed desired behavior per textView and even per same completion session.
                 // The right fix would be to send this information as a result of the method. 
                 // Then, the Editor would choose the right behavior for mixed cases.
-                _textView.Options.GlobalOptions.SetOptionValue(s_nonBlockingCompletionEditorOption, !_editorOptionsService.GlobalOptions.GetOption(CompletionViewOptionsStorage.BlockForCompletionItems, service.Language));
-                _responsiveCompletionEnabled = _textView.Options.GetOptionValue(DefaultOptions.ResponsiveCompletionOptionId);
+                var blockForCompletionItem = _editorOptionsService.GlobalOptions.GetOption(CompletionViewOptionsStorage.BlockForCompletionItems, service.Language);
+                _textView.Options.GlobalOptions.SetOptionValue(s_nonBlockingCompletionEditorOption, !blockForCompletionItem);
```

Comment:
we have another option as well?  This now feels like 3 separate options ion this space :)  Def confusing.  I don't get what each is supposed to control..

Reply:
I don't have context on this option (CompletionViewOptionsStorage.BlockForCompletionItems), but it seems it was only used for TS and FS to make completion in their files non-blocking.

Reply:
gotcha... i would love if you could talk to them if we can remove this and they can just use the VS concepts here :)  (not blocking (no pun intended) for this pr).

Reply:
Will do